transaction/std/src/transaction_builder.rs

```diff
@@ -162,6 +174,17 @@ impl<FPR: FogPubkeyResolver> TransactionBuilder<FPR> {
             .push(InputMaterials::Signable(input_credentials));
     }
 
+    /// Add a View Only Input to the transaction.
+    ///
+    /// # Arguments
+    /// * `input_view_only_materials` - Materials required to construct the
+    ///   onetime private key for the input when combined with the spend private
+    ///   key for the account.
```

Comment:
Maybe this will help make it a bit clear for future people looking at this.

```suggestion
    /// * `input_view_only_materials` - Materials required to construct the
    ///   onetime private key for the input, excluding the spend private
    ///   key for the account.
    ///
    /// Using this method prevents you from fully building a transaction, since the spend key material is
    /// unavailable. However, it allows you to call `get_signing_data` which returns all the information about
    /// the transaction that allows it to later be signed by an external tool that have access to the spend private
    /// key.
    /// For an example of how this can be used to generate a fully valid and signed transaction, please see unit test below.
```
And as this comment mentions, there should be a unit test that takes the signing data + a spend private key, and uses it to construct a valid, signed transaction. I am not sure how hard it is to write such a test...

Reply:
Hmm... That should be doable, I'll give it a shot.